spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/ModuleCommands.java

```diff
@@ -191,14 +208,48 @@ public Object getValue(int row, int column) {
 				int currentRow = row - 1;
 				if (mappings.get(key).size() > currentRow) {
 					return mappings.get(key).get(currentRow);
-				} else {
+				}
+				else {
 					return null;
 				}
 			}
 		};
 		return DataFlowTables.applyStyle(new TableBuilder(model)).build();
 	}
 
+	@CliCommand(value = IMPORT_MODULES, help = "Register all modules listed in a properties file")
+	public String importFromResource(
+			@CliOption(mandatory = true,
+					key = {"", "uri"},
+					help = "URI for the properties file")
+			String uri,
+			@CliOption(key = "local",
+				help = "whether to resolve the URI locally (as opposed to on the server)",
+				specifiedDefaultValue = "true",
+				unspecifiedDefaultValue = "true")
+			boolean local,
+			@CliOption(key = "force",
+				help = "force update if any module already exists (only if not in use)",
+				specifiedDefaultValue = "true",
+				unspecifiedDefaultValue = "false")
+			boolean force) {
+		if (local) {
+			try {
+				Resource resource = this.resourceLoader.getResource(uri);
+				Properties modules = PropertiesLoaderUtils.loadProperties(resource);
+				moduleOperations().registerAll(modules, force);
+				return String.format("Successfully registered modules: " + modules.keySet());
```

Comment:
I agree. It's not trivial to display everything (since it's a paged resources return value). At least, in the case where the URI is resolved within the shell process, the app keys are known, hence I decided to show that information (whereas the server-side resolution of the URI now at least shows the number of apps registered per your suggestion). I didn't want to go much further for now since the follow-up work on error handling might affect what we display.